Minio/V4Authenticator.cs

```diff
@@ -463,11 +460,13 @@ private void SetHostHeader(IRestRequest request, string hostUrl)
         private void SetContentSha256(IRestRequest request)
         {
             // No need to compute SHA256 if endpoint scheme is https
-            if (this.scheme.Equals("https"))
+            if (isSecure && !isAnonymous)
             {
                 request.AddHeader("x-amz-content-sha256","UNSIGNED-PAYLOAD");
                 return;
             }
+            if (this.isAnonymous)
+                return;
```

Comment:
You can simplify the code by reordering a bit: put the above if block as the first one in the function, then adding the unsigned-payload header if condition may just check if isSecure is true.